forms/react-hook-form.md

```diff
@@ -0,0 +1,307 @@
+
+1\. Install Yup:
+
+```bash
+pnpm install @hookform/resolvers yup
+```
+
+2\.  Import necessary modules:
+
+In your form component, import the required modules from Yup and React Hook Form.
+
+```ts
+import { FC } from 'react';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+```
+
+3\. Define Yup schema for validation:
+
+Create a Yup schema that defines the validation rules for your form fields.
+
+```javascript
+const schema = yup.object().shape({
+  name: yup.string().required('Name is required'),
+  email: yup.string().email('Invalid email').required('Email is required'),
+  age: yup.number().integer().positive().required('Age is required'),
+});
+```
+
+4\. Create the form component:
+
+Use the `useForm` hook and the `yupResolver` from `@hookform/resolvers/yup` to integrate Yup validation with React Hook Form.
+
+```ts
+const MyFormComponent:FC = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+
+  const onSubmit = (data) => {
+    console.log(data); // Data contains the form input values
+  };
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <label htmlFor="name">Name:</label>
+      <input type="text" id="name" {...register('name')} />
+      {errors.name && <p>{errors.name.message}</p>}
+
+      <label htmlFor="email">Email:</label>
+      <input type="text" id="email" {...register('email')} />
+      {errors.email && <p>{errors.email.message}</p>}
+
+      <label htmlFor="age">Age:</label>
+      <input type="number" id="age" {...register('age')} />
+      {errors.age && <p>{errors.age.message}</p>}
+
+      <button type="submit">Submit</button>
+    </form>
+  );
+}
+```
+
+5\. Submit the form:
+
+When the form is submitted, the `handleSubmit` function will automatically trigger the Yup validation defined in the schema. If there are any validation errors, they will be displayed next to the respective form fields.
+
+With Yup and React Hook Form integrated, your form will have robust validation based on the Yup schema, providing a smooth user experience.
+
+Yup's documentation is a great resource for exploring all the available validation rules and methods: https://github.com/jquense/yup.
+
+## Validation with Zod
+
+Validation with Zod is similar to Yup. To use Zod with React Hook Form, you can follow these steps:
+
+
+1\. Install Zod:
+
+```bash
+pnpm install @hookform/resolvers zod
+```
+
+2\. Create a Zod schema that defines the validation rules for your form fields.
+
+```javascript
+import { z } from 'zod';
+
+const schema = z.object({
+  name: z.string().nonempty('Name is required'),
+  email: z.string().email('Invalid email').nonempty('Email is required'),
+  age: z.number().int('Age must be an integer').nonnegative('Age is required'),
+});
+```
```

Comment:
The same here. I'd recommend adding more complex examples. I will share them with you